test/examples/uct_hello_world.c

```diff
@@ -1,17 +1,611 @@
 /**
-* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
-*
+* Copyright (C) UT-Battelle, LLC. 2015. ALL RIGHTS RESERVED.
 * See file LICENSE for terms.
 */
 
-#define HAVE_CONFIG_H /* Force using config.h, so test would fail if header
-                         actually tries to use it */
+#include "ucx_hello_world.h"
+
 #include <uct/api/uct.h>
-#include <stdio.h>
+#include <ucs/type/status.h>
+#include <ucs/debug/log.h>
 
+typedef enum {
+    FUNC_AM_SHORT,
+    FUNC_AM_BCOPY,
+    FUNC_AM_ZCOPY
+} func_am_t;
 
-int main(int argc, char **argv)
+typedef struct {
+    char        *server_name;
+    uint16_t    server_port;
+    func_am_t   func_am_type;
+    const char  *dev_name;
+    const char  *tl_name;
+    long        test_strlen;
+} cmd_args_t;
+
+static char *func_am_t_str(func_am_t func_am_type)
 {
-    printf("Hello World\n");
+    switch (func_am_type) {
+    case FUNC_AM_SHORT:
+        return "uct_ep_am_short";
+    case FUNC_AM_BCOPY:
+        return "uct_ep_am_bcopy";
+    case FUNC_AM_ZCOPY:
+        return "uct_ep_am_zcopy";
+    }
+    return NULL;
+}
+
+static size_t func_am_max_size(func_am_t func_am_type,
+                               const uct_iface_attr_t *attr)
+{
+    switch (func_am_type) {
+    case FUNC_AM_SHORT:
+        return attr->cap.am.max_short;
+    case FUNC_AM_BCOPY:
+        return attr->cap.am.max_bcopy;
+    case FUNC_AM_ZCOPY:
+        return attr->cap.am.max_zcopy;
+    }
     return 0;
 }
+
+struct iface_info {
```

Comment:
perhaps it is better to be aligned with previous structs style and use `typedef struct iface_info {} iface_info_t;`